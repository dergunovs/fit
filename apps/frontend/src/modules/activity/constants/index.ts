import { TActivityChartType } from 'fitness-tracker-contracts';

export const URL_ACTIVITY = '/activities';
export const URL_ACTIVITY_CREATE = `${URL_ACTIVITY}/create`;
export const URL_ACTIVITY_EDIT = `${URL_ACTIVITY}/edit`;

export const URL_ACTIVITY_ADMIN = '/activities_admin';
export const URL_ACTIVITY_ADMIN_EDIT = `${URL_ACTIVITY_ADMIN}/edit`;

export const ACTIVITY_LIST_HEADERS = [{ title: 'Создано' }, { title: 'Пользователь' }];

export const CHART_TYPES: { title: string; value: TActivityChartType }[] = [
  { title: 'Занятия', value: 'activity' },
  { title: 'Подходы', value: 'set' },
  { title: 'Повторы', value: 'repeat' },
  { title: 'Мышцы', value: 'muscle' },
];

export const ACTIVITY_STATISTICS_GAP = 30;
