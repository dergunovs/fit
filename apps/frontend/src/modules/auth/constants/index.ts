export const URL_SETUP = '/setup';

export const TOKEN_NAME = 'fit_token';

export const AUTH_FORM_HEADER_LOGIN = 'Логин';
export const AUTH_FORM_HEADER_SETUP = 'Создать администратора';

export const AUTH_FORM_SUBMIT_BUTTON_LOGIN = 'Войти';
export const AUTH_FORM_SUBMIT_BUTTON_SETUP = 'Создать';
