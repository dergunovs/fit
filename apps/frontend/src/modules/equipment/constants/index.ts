export const URL_EQUIPMENT = '/equipment';
export const URL_EQUIPMENT_CREATE = `${URL_EQUIPMENT}/create`;
export const URL_EQUIPMENT_EDIT = `${URL_EQUIPMENT}/edit`;

export const EQUIPMENT_LIST_HEADERS = [{ title: 'Название' }];
