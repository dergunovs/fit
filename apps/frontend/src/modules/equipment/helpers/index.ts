import { IEquipment } from 'fitness-tracker-contracts';

export function filterEquipmentByWeights(equipments?: IEquipment[], isWeights?: boolean) {
  if (!equipments?.length) return [];

  return equipments?.filter((equipment) => (isWeights ? equipment.isWeights : !equipment.isWeights));
}
