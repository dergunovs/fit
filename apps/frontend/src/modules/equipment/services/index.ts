import { ComputedRef } from 'vue';
import {
  API_EQUIPMENT,
  TGetEquipmentsDTO,
  TGetEquipmentDTO,
  TPostEquipmentDTO,
  TPostEquipmentDataDTO,
  TUpdateEquipmentDTO,
  TUpdateEquipmentDataDTO,
  TDeleteEquipmentDTO,
} from 'fitness-tracker-contracts';
import { useMutation, useQuery, api } from 'mhz-helpers';

export const equipmentService = {
  getAll: () =>
    useQuery({
      queryKey: [API_EQUIPMENT],
      queryFn: async () => {
        const { data } = await api.get<TGetEquipmentsDTO>(API_EQUIPMENT);

        return data.data;
      },
    }),

  getOne: (options: object, id: ComputedRef<string>) =>
    useQuery({
      queryKey: [API_EQUIPMENT, id],
      queryFn: async () => {
        const { data } = await api.get<TGetEquipmentDTO>(`${API_EQUIPMENT}/${id.value}`);

        return data.data;
      },
      ...options,
    }),

  create: (options: object) =>
    useMutation({
      mutationKey: [API_EQUIPMENT],
      mutationFn: async (formData: TPostEquipmentDataDTO) => {
        const { data } = await api.post<TPostEquipmentDTO>(API_EQUIPMENT, formData);

        return data;
      },
      ...options,
    }),

  update: (options: object) =>
    useMutation({
      mutationKey: [API_EQUIPMENT],
      mutationFn: async (formData: TUpdateEquipmentDataDTO) => {
        const { data } = await api.patch<TUpdateEquipmentDTO>(`${API_EQUIPMENT}/${formData._id}`, formData);

        return data;
      },
      ...options,
    }),

  delete: (options: object) =>
    useMutation({
      mutationKey: [API_EQUIPMENT],
      mutationFn: async (id: string) => {
        const { data } = await api.delete<TDeleteEquipmentDTO>(`${API_EQUIPMENT}/${id}`);

        return data;
      },
      ...options,
    }),
};
