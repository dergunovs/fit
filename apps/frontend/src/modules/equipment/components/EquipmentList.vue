<template>
  <UiTable :headers="EQUIPMENT_LIST_HEADERS" :isLoading="!props.equipments?.length" :lang="locale">
    <tr v-for="equipment in props.equipments" :key="equipment._id" data-test="equipment-table-row">
      <td data-grow>
        <RouterLink :to="`${URL_EQUIPMENT_EDIT}/${equipment._id}`" data-test="equipment-table-title-link">
          {{ equipment.title }}<span v-if="equipment.title_en"> ({{ equipment.title_en }})</span>
        </RouterLink>
      </td>
    </tr>
  </UiTable>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import { DefaultLocaleMessageSchema, useI18n } from 'vue-i18n';
import { UiTable } from 'mhz-ui';
import { IEquipment, TLocale } from 'fitness-tracker-contracts';

import { URL_EQUIPMENT_EDIT } from '@/equipment/constants';

interface IProps {
  equipments?: IEquipment[];
}

const props = defineProps<IProps>();

const { t, locale } = useI18n<DefaultLocaleMessageSchema, TLocale>();

const EQUIPMENT_LIST_HEADERS = computed(() => [{ title: t('title') }]);
</script>
