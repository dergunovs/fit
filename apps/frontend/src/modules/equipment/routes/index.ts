import { URL_EQUIPMENT, URL_EQUIPMENT_CREATE, URL_EQUIPMENT_EDIT } from '@/equipment/constants';

export const equipmentRoutes = [
  {
    path: URL_EQUIPMENT,
    name: 'EquipmentList',
    component: () => import('@/equipment/pages/EquipmentListPage.vue'),
    meta: { layout: 'admin' },
  },
  {
    path: URL_EQUIPMENT_CREATE,
    name: 'EquipmentCreate',
    component: () => import('@/equipment/pages/EquipmentCreatePage.vue'),
    meta: { layout: 'admin' },
  },
  {
    path: `${URL_EQUIPMENT_EDIT}/:equipment`,
    name: 'EquipmentEdit',
    component: () => import('@/equipment/pages/EquipmentEditPage.vue'),
    meta: { layout: 'admin' },
  },
];
