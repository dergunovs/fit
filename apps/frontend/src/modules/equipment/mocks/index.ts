import { vi } from 'vitest';
import {
  TDeleteEquipmentDTO,
  TPostEquipmentDataDTO,
  TPostEquipmentDTO,
  TUpdateEquipmentDataDTO,
  TUpdateEquipmentDTO,
} from 'fitness-tracker-contracts';

import { equipmentService } from '@/equipment/services';
import { mockMutationReply, mockQueryReply } from '@/common/mocks';
import { EQUIPMENT_FIXTURE, EQUIPMENTS_FIXTURE } from '@/equipment/fixtures';
import { IOnSuccess } from '@/common/interface';

const spyGetEquipment = vi.spyOn(equipmentService, 'getOne').mockReturnValue(mockQueryReply(EQUIPMENT_FIXTURE));

const spyGetEquipments = vi
  .spyOn(equipmentService, 'getAll')
  .mockImplementation(() => mockQueryReply(EQUIPMENTS_FIXTURE));

const spyCreateEquipment = vi.fn();
const spyUpdateEquipment = vi.fn();
const spyDeleteEquipment = vi.fn();

const mockOnSuccess: IOnSuccess = {
  create: undefined,
  update: undefined,
  delete: undefined,
};

vi.spyOn(equipmentService, 'create').mockImplementation((options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) mockOnSuccess.create = options.onSuccess;

  return mockMutationReply<TPostEquipmentDTO, TPostEquipmentDataDTO>(spyCreateEquipment);
});

vi.spyOn(equipmentService, 'update').mockImplementation((options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) mockOnSuccess.update = options.onSuccess;

  return mockMutationReply<TUpdateEquipmentDTO, TUpdateEquipmentDataDTO>(spyUpdateEquipment);
});

vi.spyOn(equipmentService, 'delete').mockImplementation((options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) mockOnSuccess.delete = options.onSuccess;

  return mockMutationReply<TDeleteEquipmentDTO, string>(spyDeleteEquipment);
});

export { spyGetEquipment, spyGetEquipments, spyCreateEquipment, spyUpdateEquipment, spyDeleteEquipment, mockOnSuccess };
