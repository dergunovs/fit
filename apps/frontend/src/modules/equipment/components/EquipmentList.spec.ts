import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import EquipmentList from './EquipmentList.vue';

import { wrapperFactory } from '@/common/test';
import { EQUIPMENTS_FIXTURE } from '@/equipment/fixtures';
import { URL_EQUIPMENT_EDIT } from '@/equipment/constants';

const equipmentTableRow = dataTest('equipment-table-row');
const equipmentTableTitleLink = dataTest('equipment-table-title-link');

let wrapper: VueWrapper<InstanceType<typeof EquipmentList>>;

beforeEach(() => {
  wrapper = wrapperFactory(EquipmentList, { equipments: EQUIPMENTS_FIXTURE });
});

enableAutoUnmount(afterEach);

describe('EquipmentList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(EquipmentList)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows equipments in table', async () => {
    expect(wrapper.findAll(equipmentTableRow).length).toBe(EQUIPMENTS_FIXTURE.length);
    expect(wrapper.find(equipmentTableTitleLink).text()).toBe(EQUIPMENTS_FIXTURE[0].title);
  });

  it('sets equipment page link', async () => {
    expect(wrapper.find(equipmentTableTitleLink).attributes('to')).toBe(
      `${URL_EQUIPMENT_EDIT}/${EQUIPMENTS_FIXTURE[0]._id}`
    );
  });
});
