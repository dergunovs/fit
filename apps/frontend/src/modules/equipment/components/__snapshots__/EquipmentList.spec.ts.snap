// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EquipmentList > matches snapshot 1`] = `
"<table headers="[object Object]" isloading="false" lang="ru">
  <tbody>
    <tr data-test="equipment-table-row">
      <td data-grow=""><a to="/equipment/edit/147456456" data-test="equipment-table-title-link">Гантели
          <!--v-if-->
        </a></td>
    </tr>
    <tr data-test="equipment-table-row">
      <td data-grow=""><a to="/equipment/edit/245645645645" data-test="equipment-table-title-link">Скамья
          <!--v-if-->
        </a></td>
    </tr>
  </tbody>
</table>"
`;
